src/provider/graphql.ts

```diff
@@ -0,0 +1,432 @@
+import { C, fromHex } from "../mod.ts";
+import {
+  Address,
+  Assets,
+  CostModels,
+  Delegation,
+  OutRef,
+  ProtocolParameters,
+  Provider,
+  Slot,
+  Transaction,
+  TxHash,
+  Unit,
+  UTxO,
+} from "../types/mod.ts";
+
+export class GraphQL implements Provider {
+  gqlUrl: string;
+  submitUrl: string;
+  authToken: string | undefined;
+  constructor(gqlUrl: string, submitUrl: string, authToken?: string) {
+    this.gqlUrl = gqlUrl;
+    this.submitUrl = submitUrl;
+    if (authToken) this.authToken = authToken;
+  }
+
+  async getProtocolParameters(): Promise<ProtocolParameters> {
+    const ProtocolParametersQuery = `
+    query getProtocolParameters {
+      cardano {
+        tip {
+          slotNo
+        }
+        currentEpoch {
+          protocolParams {
+            minFeeA
+            minFeeB
+            poolDeposit
+            keyDeposit
+            coinsPerUtxoByte
+            maxValSize
+            maxTxSize
+            priceMem
+            priceStep
+            maxTxExMem
+            maxTxExSteps
+            collateralPercent
+            maxCollateralInputs
+            costModels
+          }
+        }
+      }
+    }`;
+    const fullGraphqlQuery = {
+      "operationName": "getProtocolParameters",
+      "query": ProtocolParametersQuery,
+      "variables": {},
+    };
+    const qdata = await this.queryGraphQL(fullGraphqlQuery);
+    const params: ProtocolParamsGQL =
+      qdata.data.cardano.currentEpoch.protocolParams;
+    return {
+      minFeeA: parseInt(params.minFeeA.toString()),
+      minFeeB: parseInt(params.minFeeB.toString()),
+      maxTxSize: parseInt(params.maxTxSize.toString()),
+      maxValSize: parseInt(params.maxValSize.toString()),
+      keyDeposit: BigInt(params.keyDeposit),
+      poolDeposit: BigInt(params.poolDeposit),
+      priceMem: parseFloat(params.priceMem.toString()),
+      priceStep: parseFloat(params.priceStep.toString()),
+      coinsPerUtxoByte: BigInt(params.coinsPerUtxoByte),
+      maxTxExMem: BigInt(params.maxTxExMem),
+      maxTxExSteps: BigInt(params.maxTxExSteps),
+      collateralPercentage: parseInt(params.collateralPercent.toString()),
+      maxCollateralInputs: parseInt(params.maxCollateralInputs.toString()),
+      costModels: params.costModels,
+    };
+  }
+
+  async getCurrentSlot(): Promise<Slot> {

```

Comment:
This endpoint is not necessary and not a part of the provider interface anymore. The current slot can be calculate off-chain.